src/main.rs

```diff
@@ -0,0 +1,449 @@
+use astar_primitives::evm::EVM_REVERT_CODE;
+use astar_primitives::genesis::GenesisAccount;
+use astar_primitives::oracle::CurrencyAmount;
+use astar_primitives::{AccountId, Balance};
+use astar_runtime::{
+    AllPalletsWithSystem, Balances, CommunityTreasuryPalletId, Executive, ParachainInfo,
+    Precompiles, Runtime, RuntimeCall, RuntimeOrigin, TreasuryPalletId, UncheckedExtrinsic,
+    SLOT_DURATION,
+};
+use codec::{DecodeLimit, Encode};
+use cumulus_primitives_core::relay_chain::HeadData;
+use cumulus_primitives_core::relay_chain::Header as RelayHeader;
+use cumulus_test_relay_sproof_builder::RelayStateSproofBuilder;
+use frame_support::dispatch::GetDispatchInfo;
+use frame_support::traits::IntegrityTest;
+use frame_support::weights::constants::WEIGHT_REF_TIME_PER_SECOND;
+use frame_support::weights::Weight;
+use frame_system::Account;
+use pallet_balances::{Holds, TotalIssuance};
+use pallet_dapp_staking::Ledger;
+use sp_consensus_aura::{Slot, AURA_ENGINE_ID};
+use sp_core::sr25519::Public;
+use sp_core::H256;
+use sp_runtime::traits::{AccountIdConversion, Dispatchable, Get, Header};
+use sp_runtime::{Digest, DigestItem, Storage};
+use sp_state_machine::BasicExternalities;
+use std::iter;
+use std::time::{Duration, Instant};
+
+fn main() {
+    let alice = GenesisAccount::<Public>::from_seed("Alice");
+    let bob = GenesisAccount::<Public>::from_seed("Bob");
+    let charlie = GenesisAccount::<Public>::from_seed("Charlie");
+    let dave = GenesisAccount::<Public>::from_seed("Dave");
+    let eve = GenesisAccount::<Public>::from_seed("Eve");
+
+    let accounts = vec![&alice, &bob, &charlie, &dave, &eve];
+
+    let genesis = generate_genesis(&accounts);
+
+    ziggy::fuzz!(|data: &[u8]| {
+        process_input(&accounts, &genesis, data);
+    });
+}
+
+fn generate_genesis(accounts: &Vec<&GenesisAccount<Public>>) -> Storage {
```

Comment:
You can consider moving this into `Astar` repo.

If I remember correctly, named genesis configs are supported now.
One such config could be `fuzzer`.